presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetPageSource.java

```diff
@@ -213,6 +214,58 @@ public Page getNextPage()
         }
     }
 
+    private Block buildNullBlock(Type type, int batchSize)
+    {
+        BlockBuilder blockBuilder = type.createBlockBuilder(new BlockBuilderStatus(), batchSize);
+        for (int j = 0; j < batchSize; j++) {
+            blockBuilder.appendNull();
+        }
+        return blockBuilder.build();
+    }
+
+    private Block buildRowBlock(List<TypeSignatureParameter> parameters, int batchSize, int level)
+    {
+        Block[] blocks = new Block[parameters.size()];
+        List<String> fieldNames = new ArrayList<>();
+        for (int i = 0; i < parameters.size(); i++) {
+            NamedTypeSignature namedTypeSignature = parameters.get(i).getNamedTypeSignature();
+            Type fieldType = typeManager.getType(namedTypeSignature.getTypeSignature());
+            fieldNames.add(namedTypeSignature.getName());
+            blocks[i] = buildNullBlock(fieldType, batchSize);
+        }
+
+        List<String> pathFields = new ArrayList<>();
+        List<String[]> paths = requestedSchema.getPaths();
+        for (int i = 0; i < paths.size(); i++) {
+            String[] path = paths.get(i);
+            if (path.length >= level + 1) {
+                String pathField = path[level];
+                if (pathFields.contains(pathField)) {
+                    continue;
+                }
+                pathFields.add(pathField);
+                int index = fieldNames.indexOf(pathField);
+                if (index != -1) {
+                    Type fieldType = typeManager.getType(parameters.get(index).getNamedTypeSignature().getTypeSignature());
+                    if (ROW.equals(fieldType.getTypeSignature().getBase())) {
+                        blocks[index] = buildRowBlock(fieldType.getTypeSignature().getParameters(), batchSize, level + 1);
+                    }
+                    else {
+                        RichColumnDescriptor columnDescriptor = getDescriptor(fileSchema, requestedSchema, pathField, level + 1);
+                        blocks[index] = new LazyBlock(batchSize, new ParquetBlockLoader(columnDescriptor, fieldType));
+                   }
+                }
+            }
+        }
+
+        InterleavedBlock interleavedBlock = new InterleavedBlock(blocks);
+        int[] offsets = new int[batchSize];
+        for (int i = 0; i < batchSize; i++) {
+            offsets[i] = (i + 1) * parameters.size();
+        }
+        return new ArrayBlock(interleavedBlock, wrappedIntArray(offsets), 0, allocate(batchSize));
```

Comment:
nullVector is treated as {fieldA=null, fieldB=null...}
unlike ORC, there is not presentBit, not so easy to set value for nullVector for lazyBlock
@nezihyigitbasi @dain what do you think?